File: android/src/main/kotlin/africa/ejara/trustdart/utils/WalletError.kt
package africa.ejara.trustdart.utils


class WalletError(
    errorCode: WalletHandlerErrorCodes,
    val errorMessage: String?,
    val errorDetails: Any?
) {
    val errorCode: String = errorCode.value
}

data class WalletValidateResponse(val isValid: Boolean, val details: WalletError)

enum class WalletHandlerErrorCodes(val value: String) {
    NoWallet("no_wallet"),
    AddressNull("address_null"),
    ArgumentsNull("arguments_null"),
    TxHashNull("txhash_null")
}

class ErrorResponse {
    companion object {
        const val argumentsNull = "[path], [coin], [mnemonic] and [passphrase] are required."
        const val privateKeyNull = "Could not generate private key."
        const val publicKeyNull = "Could not generate public key."
    }
}

File: ios/Classes/utils/WalletError.swift
//
//  WalletError.swift
//  trustdart
//
//  Created by Jay on 4/5/22.
//

import Flutter

class WalletError {
    let details: FlutterError
    required init(code: WalletHandlerErrorCodes, message: String?, details: Any?){
        self.details = FlutterError(code: code.rawValue, message: message, details: details)
    }
}

enum WalletHandlerErrorCodes: String {
    case noWallet = "no_wallet"
    case addressNull = "address_null"
    case argumentsNull = "arguments_null"
    case txHashNull = "txhash_null"
}
